fastify -r does not behave like node -r

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported


### Fastify version

3.15.1

### Node.js version

16.0.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

Big Sur

### Description

fastify -r does not behave like node -r

Maybe a feature - not a bug :S.

### Steps to Reproduce

When I start an application with fastify cli like so:

`fastify start -r '@app/config/env' -w -l info -P dist/app.js`

`
[App] Error: Cannot find module '/'
...
[App]   code: 'MODULE_NOT_FOUND',
...
`

`@app/config/env` holds the `PORT` and is required as dependency in my package.json. It is a package in a mono-repo managed with yarn workspaces.

When I do the same with an ejected server and Node.JS

```
➜  server git:(fastify) ✗ node server.js
{... Server listening at http://127.0.0.1:3000"}

thing ➜ server git:(fastify) ✗ node -r '@app/config/env' server.js
{... Server listening at http://127.0.0.1:5000"}
```

### Expected Behavior

I would expect `fastify -r ` to behave like `node -r`.

--- a/start.js
+++ b/start.js
@@ -14,7 +14,7 @@ const watch = require('./lib/watch')
 const parseArgs = require('./args')
 const {
   exit,
-  requirePath,
+  requireModule,
   requireFastifyForModule,
   requireServerPluginFromPath,
   showHelpForCommand
@@ -72,7 +72,7 @@ async function runFastify (args) {
           /* This check ensures we ignore `-r ""`, trailing `-r`, or
            * other silly things the user might (inadvertently) be doing.
            */
-          requirePath(module)
+          requireModule(module)
         }
       })
     } catch (e) {
@@ -94,7 +94,7 @@ async function runFastify (args) {
   let logger
   if (opts.loggingModule) {
     try {
-      logger = requirePath(opts.loggingModule)
+      logger = requireModule(opts.loggingModule)
     } catch (e) {
       module.exports.stop(e)
     }

--- a/test/start.test.js
+++ b/test/start.test.js
@@ -821,6 +821,24 @@ test('should support custom logger configuration', async t => {
   t.pass('server closed')
 })
 
+test('preloading a built-in module works', async t => {
+  t.plan(1)
+
+  const argv = ['-r', 'path', './examples/plugin.js']
+  const fastify = await start.start(argv)
+  await fastify.close()
+  t.pass('server closed')
+})
+
+test('preloading a module in node_modules works', async t => {
+  t.plan(1)
+
+  const argv = ['-r', 'tap', './examples/plugin.js']
+  const fastify = await start.start(argv)
+  await fastify.close()
+  t.pass('server closed')
+})
+
 test('should throw on logger configuration module not found', async t => {
   t.plan(2)
 

--- a/util.js
+++ b/util.js
@@ -19,12 +19,15 @@ function exit (message) {
   process.exit()
 }
 
-function requirePath (filePath) {
-  const moduleFilePath = path.resolve(filePath)
-  const moduleFileExtension = path.extname(filePath)
-  const modulePath = moduleFilePath.split(moduleFileExtension)[0]
-  const module = require(modulePath)
-  return module
+function requireModule (moduleName) {
+  if (fs.existsSync(moduleName)) {
+    const moduleFilePath = path.resolve(moduleName)
+    const moduleFileExtension = path.extname(moduleName)
+    const modulePath = moduleFilePath.split(moduleFileExtension)[0]
+    return require(modulePath)
+  } else {
+    return require(moduleName)
+  }
 }
 
 function requireFastifyForModule (modulePath) {
@@ -95,4 +98,4 @@ function showHelpForCommand (commandName) {
   }
 }
 
-module.exports = { exit, requirePath, requireFastifyForModule, showHelpForCommand, requireServerPluginFromPath }
+module.exports = { exit, requireModule, requireFastifyForModule, showHelpForCommand, requireServerPluginFromPath }

